### DISCUSSION TITLE
Adding SSO to directus?

### DISCUSSION THREAD

**Discussion by HashemKhalifa** [directus/directus#3886] [Archive]
Hello,

I would like to implement a custom SSO with directus9! the documentation seems to be broken with 404-page error.

is there's any idea how I can implement this?

**Comment by benhaynes** [comment]
Thanks! We'll be making sure this (and other) Docs are up-to-date over the next few days! Ref: @rijkvanzanten

**Reply by HashemKhalifa** [reply]
Thank you @benhaynes for your quick reply, here is an example https://docs.directus.io/reference/api/authentication.html

**Reply by doodlemania2** [reply]
I was able to get Microsoft OAuth up and working but running into this issue: https://github.com/directus/directus/pull/3827 so can't quite get logged in yet.

Also, on this doc, a suggestion:
https://docs.directus.io/reference/environment-variables.html#oauth

OAUTH_PROVIDERS ="microsoft"
OAUTH_MICROSOFT_KEY = "<microsoft_application_id>"
OAUTH_MICROSOFT_SECRET = "<microsoft_application_secret_key>"
OAUTH_MICROSOFT_SCOPE = "openid email"
OAUTH_MICROSOFT_AUTHORIZE_URL = "https://login.microsoftonline.com/<microsoft_**TENANT**_id>/oauth2/v2.0/authorize"
OAUTH_MICROSOFT_ACCESS_URL = "https://login.microsoftonline.com/<microsoft_**TENANT**_id>/oauth2/v2.0/token"

PUBLIC_URL = "<public_url_of_directus_instance>" **### Should be https://public_url/**

**Reply by ederduran** [reply]
@doodlemania2  
Me neither, were you able to integrate it successfully?
